src/main/webapp/app/exercises/programming/shared/instructions-render/extensions/programming-exercise-task.extension.ts

```diff
@@ -14,6 +14,8 @@ import { ExerciseHint } from 'app/entities/hestia/exercise-hint.model';
 
 @Injectable({ providedIn: 'root' })
 export class ProgrammingExerciseTaskExtensionWrapper implements ArtemisShowdownExtensionWrapper {
+    viewContainerRef: ViewContainerRef;
```

Comment:
I assume that this is an injected property like ApplicationRef?
If that is the case why did you not put it into the constructor parameter list?
If that is not the case consider adding a comment explaining how this is created.

Reply:
That's because it fails to auto-inject:
```
R3InjectorError(DynamicTestModule)[ProgrammingExerciseTaskExtensionWrapper -> ViewContainerRef -> ViewContainerRef]: 
  NullInjectorError: No provider for ViewContainerRef!
```